cluster/manifests/kube-node-ready/daemonset.yaml

```diff
@@ -25,6 +25,15 @@ spec:
       annotations:
         logging/destination: "{{.Cluster.ConfigItems.log_destination_infra}}"
     spec:
+      # consider tagging ASG based instance with a label such that a simple
+      # nodeSelector can be used which is more performant.
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: karpenter.sh/capacity-type
```

Comment:
Let's think about if it makes sense to have our own, e.g. `zalando.org/node-provisioner=[cluster-autoscaler|karpenter]`. This is more descriptive and solves the point mentioned in the comment.

Reply:
`karpenter.sh/initialized` or `karpenter.sh/provisioner-name` might also be good candidates.